### ISSUE TITLE
Scheme cog-bind-example.scm exception

### ISSUE THREAD

**Issue by pennachin** [opencog/atomspace#280]
Running `cog-bind-example.scm` gives me this:

```
opencog@a2a7a0c3800d:/atomspace/examples/guile$ cat cog-bind-example.scm | guile

ERROR: In procedure cog-new-link:
ERROR: Throw to key `C++-EXCEPTION' with args `("cog-new-link" "Expected a VariableNode or a TypedVariableLink, got: ListLink (/tmp/atomspace-master/opencog/atoms/core/VariableList.cc:48)")'.

Entering a new prompt.  Type `,bt' for a backtrace or `,q' to continue.
<unnamed port>:96:0: In procedure #<procedure 2f7c480 at <current input>:96:0 ()>:
<unnamed port>:96:0: In procedure #<procedure 2f7c480 at <current input>:96:0 ()>: Unbound variable: x

Entering a new prompt.  Type `,bt' for a backtrace or `,q' to continue.
scheme@(guile-user) [2]> 
scheme@(guile-user) [1]> 
scheme@(guile-user)> 
```

while evaluating this form:

```
(define x
	(BindLink
		(ListLink
			(VariableNode "$var0")
			(VariableNode "$var1")
			(VariableNode "$verb")
		)
		(AndLink
			(EvaluationLink
				(PredicateNode "_obj")
				(ListLink
					(VariableNode "$verb") ; (ConceptNode "make")
					(VariableNode "$var0") ; (ConceptNode "pottery")
				)
			)
			(EvaluationLink
				(PredicateNode "from")
				(ListLink
					(VariableNode "$verb") ; (ConceptNode "make")
					(VariableNode "$var1") ; (ConceptNode "clay")
				)
			)
		)
		(EvaluationLink
			(PredicateNode "make_from")
			(ListLink
				(VariableNode "$var0")
				(VariableNode "$var1")
			)
		)
	)
)
```

I may be doing something stupid, but the other examples run without exceptions (although results differ from comments in another case)

**Comment by williampma** [comment]
I think you want `VariableList` instead of `ListLink` for the variables declaration.

**Comment by pennachin** [comment]
Thanks! I fixed it on #279
